File: VB/DxSample/FilterAlingment.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows

Namespace DxSample
	Friend Class FilterAlingment
		Public Shared ReadOnly FilterAlingmentProperty As DependencyProperty = DependencyProperty.RegisterAttached("FilterAlingment", GetType(HorizontalAlignment), GetType(FilterAlingment), New UIPropertyMetadata(Nothing))

		Public Shared Function GetFilterAlingment(ByVal target As DependencyObject) As HorizontalAlignment
			Return CType(target.GetValue(FilterAlingmentProperty), HorizontalAlignment)
		End Function

		Public Shared Sub SetFilterAlingment(ByVal target As DependencyObject, ByVal value As HorizontalAlignment)
			target.SetValue(FilterAlingmentProperty, value)
		End Sub
	End Class
End Namespace

File: CS/DxSample/FilterAlingment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;

namespace DxSample
{
    class FilterAlingment
    {
        public static readonly DependencyProperty FilterAlingmentProperty = DependencyProperty.RegisterAttached("FilterAlingment", typeof(HorizontalAlignment), typeof(FilterAlingment), new UIPropertyMetadata(null));

        public static HorizontalAlignment GetFilterAlingment(DependencyObject target)
        {
            return (HorizontalAlignment)target.GetValue(FilterAlingmentProperty);
        }

        public static void SetFilterAlingment(DependencyObject target, HorizontalAlignment value)
        {
            target.SetValue(FilterAlingmentProperty, value);
        }  
    }
}
